### PR TITLE
Target SDK 31

### PR DESCRIPTION
**Pull request by zello-eugene**
#21
#24

- Update target SDK version to 31
- Update dependencies
- Support developing on M1 Mac
  - Silence a number of warnings

_Ignore whitespace changes **ON**_

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,25 +1,28 @@
 buildscript {
-	ext.kotlin_version = '1.4.32'
-	ext.dokka_version = '0.9.17'
+	ext.kotlin_version = '1.7.0'
+	ext.kotlin_coroutines_version = '1.6.3'
+	ext.dokka_version = '1.7.0'
```
Comment (zello-eugene) [inline build.gradle:4]: Dokka has gone a long way since we looked at the SDK the last time.

File: build.gradle [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
 	}
 }
 
-apply plugin: 'org.jetbrains.dokka-android'
+apply plugin: 'org.jetbrains.dokka'
 
 allprojects {
 	repositories {
 		google()
-		jcenter()
```
Comment (zello-eugene) [inline build.gradle:22]: JCenter is gone.

File: channel-sdk/src/main/java/com/zello/channel/sdk/SessionContextAndroid.kt [hunk 3]
```diff
@@ -32,7 +33,7 @@ internal class SessionContextAndroid(context: Context) : SessionContext {
 
 	private val context: Context = context.applicationContext
 	private var logger: SessionLogger = SessionLoggerNull()
-	private var handler: Handler = Handler()
```
Comment (zello-eugene) [inline channel-sdk/src/main/java/com/zello/channel/sdk/SessionContextAndroid.kt:35]: Handler requires a looper.

File: channel-sdk/src/main/java/com/zello/channel/sdk/platform/Arrays.java [hunk 4]
```diff
@@ -1,9 +1,9 @@
 package com.zello.channel.sdk.platform;
 
-import org.jetbrains.annotations.NotNull;
+import androidx.annotation.NonNull;
 
 /**
- * @exclude Not for public use
```
Comment (zello-eugene) [inline channel-sdk/src/main/java/com/zello/channel/sdk/platform/Arrays.java:6]: `@exclude` is not supported for java files.

File: channel-sdk/build.gradle [hunk 5]
```diff
@@ -32,118 +35,125 @@ android {
...
-	}
-	task javadocJar(type: Jar, dependsOn: sdkDocumentation) {
-		archiveClassifier = 'javadoc'
-		from sdkDocumentation.outputDirectory
+	// Library for testing
+	useLibrary 'android.test.mock'
+	buildToolsVersion '32.0.0'
+}
+
+task sdkDocumentation(type: DokkaTask) {
```
Comment (zello-eugene) [inline channel-sdk/build.gradle:52]: The task needs to be declared before any other tasks that depend on it.

File: channel-sdk/build.gradle [hunk 6]
```diff
@@ -32,118 +35,125 @@ android {
...
+	dokkaSourceSets {
+		named("main") {
+			perPackageOption {
+				matchingRegex.set("com.zello.channel.sdk.platform")
+				suppress.set(true)
+			}
```
Comment (zello-eugene) [inline channel-sdk/build.gradle:58]: Package suppress option needs to go before the flags or else it's ignored for some reason.

File: channel-sdk/src/main/java/com/zello/channel/sdk/platform/Arrays.java (hunk 4 above)
Comment (cooksey) [reply channel-sdk/src/main/java/com/zello/channel/sdk/platform/Arrays.java:6]: Really? Is there any way to get Dokka to ignore this file?

File: channel-sdk/build.gradle (hunk 6 above)
Comment (cooksey) [reply channel-sdk/build.gradle:58]: Oh cool, so does this get Dokka to ignore the Java files?

File: channel-sdk/src/main/java/com/zello/channel/sdk/location/AndroidLocationManager.kt [hunk 7]
```diff
@@ -29,8 +29,9 @@ internal class AndroidLocationManagerImpl(private val locationManager: android.l
 	}
 
 	// We only call this method if the permission has been granted. Check is upstream.
+	@Suppress("DEPRECATION")
```
Comment (cooksey) [inline channel-sdk/src/main/java/com/zello/channel/sdk/location/AndroidLocationManager.kt:32]: Should we make an issue to stop using deprecated calls?

**Review by cooksey — APPROVED** [review]
it all looks okay to me, but I'd prefer if somebody with more Android expertise can also take a look at it.

File: channel-sdk/src/main/java/com/zello/channel/sdk/platform/Arrays.java (hunk 4 above)
Comment (zello-eugene) [reply channel-sdk/src/main/java/com/zello/channel/sdk/platform/Arrays.java:6]: Yeah, we can use `perPackageOption` below.

File: channel-sdk/build.gradle (hunk 6 above)
Comment (zello-eugene) [reply channel-sdk/build.gradle:58]: Yep. The only way I was able to find.

File: channel-sdk/src/main/java/com/zello/channel/sdk/location/AndroidLocationManager.kt (hunk 7 above)
Comment (zello-eugene) [reply channel-sdk/src/main/java/com/zello/channel/sdk/location/AndroidLocationManager.kt:32]: Yep, location code is rather outdated.

Comment (zello-eugene) [reply channel-sdk/src/main/java/com/zello/channel/sdk/location/AndroidLocationManager.kt:32]: https://github.com/zelloptt/zello-android-sdk/issues/27

**Pull request merged**